sdcm/cluster.py

```diff
@@ -1967,24 +1969,28 @@ def is_scylla_installed(self):
             raise ValueError(f"Unsupported Linux distribution: {self.distro}")
         return result.exit_status == 0
 
-    def get_scylla_version(self):
-        version_commands = ["scylla --version", "rpm -q {}".format(self.scylla_pkg())]
-        for version_cmd in version_commands:
-            try:
-                result = self.remoter.run(version_cmd)
-                self.log.info("'scylla --version' output: %s", result.stdout)
-            except Exception as ex:  # pylint: disable=broad-except
-                self.log.error('Failed getting scylla version: %s', ex)
-            else:
-                match = re.match(r"((\d+)\.(\d+)\.([\d\w]+)\.?([\d\w]+)?).*", result.stdout)
-                if match:
-                    scylla_version = match.group(1)
-                    self.log.info("Found ScyllaDB version: %s", scylla_version)
-                    self.scylla_version = scylla_version
-                    return scylla_version
-                else:
-                    self.log.error("Unmatched ScyllaDB version, not caching it")
-        return None
+    def get_scylla_version(self) -> Optional[str]:
+        commands: list = ["scylla --version", ]
+
+        if self.distro.is_rhel_like:
+            commands.append(f"rpm --query --queryformat '%{{VERSION}}' {self.scylla_pkg()}")
+        else:
+            commands.append(f"dpkg-query --show --showformat '${{Version}}' {self.scylla_pkg()}")
+
+        for cmd in commands:
+            result: RunnerResult = self.remoter.run(cmd, ignore_status=True)
```

Comment:
why do we need those type annotations here?

Reply:
I'm not sure I truly understood the question. Right now we're adding annotations for two reasons: provide type hints to developers/IDEs and for the future possibility to run static source checkers. I'm not really great with type annotations, so if you can provide some more details/further readings how it should looks like in our code, it will be nice.

Reply:
I don't think we set a clear definition of usage for annotation yet, 

But I think in this case it would be better to use the annotation on the function signature and not on the code calling it.

Actually it's the first time I see that annotation is used like that, maybe we just need to get used to it.

the main issue with annotation right now, is that we still have a few branches with python, and we need to be careful with those backports, once we'll be done with those branches (or fast-forward them to a python3 version), maybe we could use it a bit more.

Reply:
There is a recommendation to use type annotations for a new code in our code best practices. That's all for now.

In my case it is not possible to modify function because it's a part of external module (`invoke`)

Variable annotations is pretty new feature (3+ years only.) It introduced in Python 3.6 ([PEP-0526](https://www.python.org/dev/peps/pep-0526/).) 

We have many Python 3 features used in our code already. Bottom line for us is Python 3.6 (because of f-strings, for example)

Reply:
I don't want the code like Java code. So for now, let's use type annotations in function/method signatures

Reply:
BTW the procedure was there in the doc
![image](https://user-images.githubusercontent.com/5269241/75146762-cab05800-570c-11ea-861d-72a2ac95e8c9.png)

Reply:
> I don't want the code like Java code. So for now, let's use type annotations in function/method signatures

NP. Removed.

Reply:
> BTW the procedure was there in the doc
> ![image](https://user-images.githubusercontent.com/5269241/75146762-cab05800-570c-11ea-861d-72a2ac95e8c9.png)

Added suggestion about variable annotations to the document.